src/models/post/relationships.rs

```diff
@@ -45,6 +45,24 @@ impl PostRelationships {
         Ok(None)
     }
 
+    /// Determines whether a post is a root post.
+    /// # Arguments
+    /// - `author_id` - A string slice representing the ID of the author of the post.
+    /// - `post_id` - A string slice representing the unique identifier of the post being checked.
+    /// # Returns
+    /// - `Ok(true)` - If the post is a root post (i.e., it has not been replied to or reposted).
+    /// - `Ok(false)` - If the post is not a root post (e.g., it has replies, reposts, or does not exist).
+    /// - `Err(DynError)` - If an error occurs while retrieving or processing the post data.
+    pub async fn is_root(author_id: &str, post_id: &str) -> Result<bool, DynError> {
+        match Self::try_from_index_json(&[author_id, post_id]).await? {
+            Some(relationship) => {
+                Ok(relationship.replied.is_none() && relationship.reposted.is_none())
```

Comment:
Reposts can be root of a thread and are indexed into `Timeline` and `TotalEngagement` sorted sets.